core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -8218,14 +8255,10 @@ private enum Clause {
     boolean shouldReplaceAliases(Config config) {
       switch (this) {
       case GROUP_BY:
-        return config.conformance().isGroupByAlias()
-                || (config.conformance().isSelectAlias()
-                != SqlConformance.SelectAliasLookup.UNSUPPORTED);
+        return config.conformance().isGroupByAlias();
```

Comment:
Previously lateral aliases in SELECT were always implying GROUP BY and HAVING aliases.
This is a change which I had introduced, but I think there is no reason these should not be separated.